### PR TITLE
fix: update model plugin

### PR DESCRIPTION
**Pull request by liuxi150**
## Summary by CodeRabbit

- **New Features**
	- Introduced new data structures for better handling of model components such as mesh, light, camera, and skybox.
	- Enhanced light component with direct inclusion of primitives, light type, color, intensity, and range specifications.
	- Updated shader ID for improved material rendering.
- **Refactor**
	- Refactored various components to utilize new and updated data structures, improving data handling and component interaction.
	- Renamed several data entities and parameters to align with the new data structure implementations.
- **Style**
	- Adjusted resource location by switching URLs, potentially altering the resource location used in the application.
- **Documentation**
	- Added type declarations for various model components enhancing code documentation and type safety.

### CODE REVIEW

File: plugin-packages/model/src/runtime/camera.ts [hunk 1]
```diff
@@ -225,8 +225,8 @@ export class PCameraManager {
    * @param owner - 所属的相机组件
```
Comment (yiiqii) [inline plugin-packages/model/src/runtime/camera.ts:225]: 注释补充

File: plugin-packages/model/src/plugin/model-item.ts [hunk 2]
```diff
@@ -259,10 +262,10 @@ export class ModelSkyboxComponent extends RendererComponent {
    * @param engine - 引擎
    * @param options - Mesh 参数
```
Comment (yiiqii) [inline plugin-packages/model/src/plugin/model-item.ts:263]: 注释的 options 要换成 data，以下同样

File: plugin-packages/model/src/runtime/light.ts [hunk 3]
```diff
@@ -48,7 +48,7 @@ export class PLight extends PEntity {
    * @param light - 灯光参数
    * @param ownerI - 所属灯光组件
```
Comment (yiiqii) [inline plugin-packages/model/src/runtime/light.ts:49]: ownerI => owner

File: plugin-packages/model/src/runtime/skybox.ts [hunk 4]
```diff
@@ -76,19 +76,33 @@ export class PSkybox extends PEntity {
...
+      for (let i = 0; i < irradianceCoeffs.length; i += 3) {
+        this.irradianceCoeffs.push([
+          irradianceCoeffs[i],
+          irradianceCoeffs[i + 1],
+          irradianceCoeffs[i + 2],
+        ]);
+      }
+    } else {
+      this.irradianceCoeffs = [];
+    }
```
[the remark this answers is not in the document]
Comment (yiiqii) [reply plugin-packages/model/src/runtime/skybox.ts:102]: @liuxi150 参考机器人的建议

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: plugin-packages/model/src/runtime/skybox.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (liuxi150) [reply plugin-packages/model/src/runtime/skybox.ts:102]: > @liuxi150 参考机器人的建议

这个写法ts上有语法错误，并且导致逻辑变复杂，不利于后续维护，我就先不采用了

File: plugin-packages/model/src/runtime/camera.ts (hunk 1 above)
Comment (liuxi150) [reply plugin-packages/model/src/runtime/camera.ts]: 已修改

File: plugin-packages/model/src/runtime/light.ts (hunk 3 above)
Comment (liuxi150) [reply plugin-packages/model/src/runtime/light.ts]: 已修改

File: plugin-packages/model/src/plugin/model-item.ts (hunk 2 above)
Comment (liuxi150) [reply plugin-packages/model/src/plugin/model-item.ts]: 已修改

**Review by yiiqii — APPROVED** [review]

**Pull request merged**